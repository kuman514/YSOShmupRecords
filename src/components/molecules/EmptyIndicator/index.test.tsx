import { cleanup, render } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import { beforeEach, describe, expect, it } from 'vitest';

import { EmptyIndicator } from '.';

describe('EmptyIndicator', () => {
  beforeEach(() => {
    cleanup();
  });

  it('should have a snapshot match for its appearance', () => {
    const routes = [
      {
        path: '/',
        element: <EmptyIndicator />,
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ['/'],
      initialIndex: 0,
    });

    const { container } = render(<RouterProvider router={router} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
