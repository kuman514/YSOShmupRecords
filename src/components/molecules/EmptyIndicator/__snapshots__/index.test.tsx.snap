// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyIndicator > should have a snapshot match for its appearance 1`] = `
<div
  class="sc-gJhJTp gsTsrA"
>
  <img
    alt="에러 발생"
    class="sc-elDIKY fzQEKb"
    src="/src/assets/images/empty.png"
  />
  <h2
    class="sc-fQpRED hKKVKD"
  >
    현재 등록된 기록이 없습니다.
  </h2>
  <button
    class="sc-braxZu breotL"
  >
    타이틀 화면으로 돌아가기
  </button>
  <button
    class="sc-braxZu breotL"
  >
    소통창구 (카카오 오픈채팅방)
  </button>
</div>
`;
